homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,18 +177,41 @@ def restore_attributes(self, last_state: State) -> None:
                 extra_state_attributes[attr] = last_state.attributes[attr]
             self._rendered[CONF_ATTRIBUTES] = extra_state_attributes
 
+    def _render_availability_template(self, variables: dict[str, Any]) -> None:
+        """Render availability template."""
+        self.error = False
+        rendered = {**self._static_rendered, **self._rendered}
+        key = CONF_AVAILABILITY
+        try:
+            if key in self._to_render_simple:
```

Comment:
Don't we need to support complex availability templates?

Reply:
According to my understanding from `__init__` it would appear that `availability`, ìcon`, `name` and `picture` can only be static or simple.

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/trigger_template_entity.py#L101-L112